llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -3949,6 +3949,20 @@ SDValue DAGCombiner::visitSUB(SDNode *N) {
       if (SDValue Result = TLI.expandABS(N1.getNode(), DAG, true))
         return Result;
 
+    // Similar to the previous rule, but this time targeting an expanded abs.
+    // (sub 0, (max X, (sub 0, X))) --> (min X, (sub 0, X))
```

Comment:
Could also do the converse:
```
(sub 0, (min X, (sub 0, X))) --> (max X, (sub 0, X))
```
but I don't know if there's a real need for it.

Reply:
I think it's fine to add it in this patch. It's done now.